apps/wallet/src/ui/app/staking/home/Stake.tsx

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useFeature } from '@growthbook/growthbook-react';
+import { useState, useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { FEATURES } from '../../experimentation/features';
+import { usePendingDelegation } from '../usePendingDelegation';
+import { ActiveDelegation } from './ActiveDelegation';
+import { DelegationCard, DelegationState } from './DelegationCard';
+import BottomMenuLayout, {
+    Menu,
+    Content,
+} from '_app/shared/bottom-menu-layout';
+import Button from '_app/shared/button';
+import { Card, CardItem } from '_app/shared/card';
+import CoinBalance from '_app/shared/coin-balance';
+import { Text } from '_app/shared/text';
+import {
+    activeDelegationIDsSelector,
+    totalActiveStakedSelector,
+} from '_app/staking/selectors';
+import Alert from '_components/alert';
+import Icon, { SuiIcons } from '_components/icon';
+import Loading from '_components/loading';
+import Overlay from '_components/overlay';
+import { useAppSelector, useObjectsState } from '_hooks';
+import { GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
+
+function StakeHome() {
+    const { loading, error, showError } = useObjectsState();
+
+    const activeDelegationIDs = useAppSelector(activeDelegationIDsSelector);
+
+    const [pendingDelegations, { isLoading: pendingDelegationsLoading }] =
+        usePendingDelegation();
+
+    const hasDelegations =
+        activeDelegationIDs.length > 0 || pendingDelegations.length > 0;
+    const [showModal, setShowModal] = useState(true);
+
+    const navigate = useNavigate();
+    const close = useCallback(() => {
```

Comment:
You shouldn't ever really need to `useCallback` (there are very rare exceptions)